tests/syntax_tests/data/printer/expr/expected/record.res.txt

```diff
@@ -100,5 +100,6 @@ type ttt = {x: int, y?: string}
 let optParen = {x: 3, y: ?(someBool ? Some("") : None)}
 let optParen = {x: 3, y: ?(3 + 4)}
 let optParen = {x: 3, y: ?foo(bar)}
-let optParen = {x: 3, y: ?foo->bar}
+let optParen = {x: 3, y: ?(foo->bar)}
```

Comment:
Is the output change intentional?

Reply:
It was random before.
We can make a choice now.

Before it was adding parens to 3+4 because it would see a printable attribute (res.optional was considered printable).
Now I have explicitly made binary operators adding parens.
I can special case `->` if without parens is preferable.
Any thoughts?

Reply:
One possibility is to treat it just like any other unary prefix operator, so have the same rules as `- e`.

Reply:
Personally I actually like it better with parens, and if that avoids special `->` here, all the better.